### PR TITLE
Upgrade spring boot and app insights

### PR DESCRIPTION
**Pull request by mkerry-moj**
Upgrade to spring-boot 3.2.4:
 - This includes an [upgrade to Spring Boot 3.2.4](https://github.com/spring-projects/spring-boot/releases/tag/v3.2.4)

Upgrade to Kotlin 1.9.23
- This includes an [upgrade to Kotlin 1.9.23](https://github.com/JetBrains/kotlin/releases/tag/v1.9.23/)

Version Upgrades:
- com.microsoft.azure:applicationinsights-agent [3.4.19 -> 3.5.1]
- com.microsoft.azure:applicationinsights-core [3.4.19 -> 3.5.1]
- io.opentelemetry:opentelemetry-api [1.32.0 -> 1.35.0]

Test dependency upgrades:
- org.mockito:mockito-junit-jupiter [5.10.0 -> 5.11.0]
- org.eclipse.jgit:org.eclipse.jgit [6.8.0.202311291450-r -> 6.9.0.202403050737-r]

### CODE REVIEW

File: release-notes/5.x.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+This includes an [upgrade to Spring Boot 3.2.4](https://github.com/spring-projects/spring-boot/releases/tag/v3.2.4)
+
+## Version Upgrades
+- com.microsoft.azure:applicationinsights-agent [3.4.19 -> 3.5.1]
+- com.microsoft.azure:applicationinsights-core [3.4.19 -> 3.5.1]
+- io.opentelemetry:opentelemetry-api [1.32.0 -> 1.36.0]
+
+## Test dependency upgrades
+- org.mockito:mockito-junit-jupiter [5.10.0 -> 5.11.0]
+
```
Comment (SimonMitchellMOJ) [inline release-notes/5.x.md:13]: there is also an update for `org.eclipse.jgit:org.eclipse.jgit [6.8.0.202311291450-r -> 6.9.0.202403050737-r]`

File: release-notes/5.x.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
+# 5.15.4
+
+## Upgrade to spring-boot 3.2.4
+This includes an [upgrade to Spring Boot 3.2.4](https://github.com/spring-projects/spring-boot/releases/tag/v3.2.4)
+
```
Comment (SimonMitchellMOJ) [inline release-notes/5.x.md:5]: would be good to include 
## Upgrade to Kotlin 1.9.23
This includes an [upgrade to Kotlin 1.9.23](https://github.com/JetBrains/kotlin/releases/tag/v1.9.23/)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/configmanagers/AppInsightsConfigManager.kt [hunk 3]
```diff
@@ -4,8 +4,8 @@ import org.gradle.api.Project
 import org.gradle.api.tasks.Copy
 import uk.gov.justice.digital.hmpps.gradle.ConfigManager
 
-private const val APP_INSIGHTS_VERSION = "3.4.19"
-const val OPENTELEMETRY_VERSION = "1.32.0"
+private const val APP_INSIGHTS_VERSION = "3.5.1"
+const val OPENTELEMETRY_VERSION = "1.36.0"
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/configmanagers/AppInsightsConfigManager.kt:8]: normally better setting the opentelemetry version to match the one in [app insights](https://github.com/microsoft/ApplicationInsights-Java/blob/ce52c7dabed16faaaf9000d30573555c77a91cf5/dependencyManagement/build.gradle.kts#L14), so in this case 1.35.0

File: build.gradle.kts [hunk 4]
```diff
@@ -50,7 +50,7 @@ gradlePlugin {
 dependencies {
   implementation(kotlin("reflect"))
 
-  implementation("org.springframework.boot:spring-boot-gradle-plugin:3.2.3")
+  implementation("org.springframework.boot:spring-boot-gradle-plugin:3.2.4")
   implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.9.22")
```
Comment (petergphillips) [inline build.gradle.kts:54]: missed out this upgrade to 1.9.23?

Comment (mkerry-hippo) [reply build.gradle.kts]: @petergphillips I've updated it to use the kotlin() function so it will just move with the kotlin version automatically

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**